src/ResourceManager/Sql/Commands.Sql/TransparentDataEncryption/Cmdlet/SetAzureRmSqlManagedInstanceTransparentDataEncryptionProtector.cs

```diff
@@ -0,0 +1,143 @@
+// ----------------------------------------------------------------------------------
+
+using Microsoft.Azure.Commands.Sql.TransparentDataEncryption.Model;
+using System.Collections.Generic;
+using System.Globalization;
+using System.Linq;
+using System.Management.Automation;
+using Microsoft.Azure.Management.Internal.Resources.Utilities.Models;
+
+namespace Microsoft.Azure.Commands.Sql.TransparentDataEncryption.Cmdlet
+{
+    /// <summary>
+    /// Defines the Set-AzureRmSqlServerTransparentDataEncryptionProtector cmdlet
+    /// </summary>
+    [Cmdlet(VerbsCommon.Set, "AzureRmSqlManagedInstanceTransparentDataEncryptionProtector", SupportsShouldProcess = true, DefaultParameterSetName = DefaultParameterSet)]
+    [Alias("Set-AzureRmSqlManagedInstanceTDEProtector")]
+    [OutputType(typeof(AzureRmSqlManagedInstanceTransparentDataEncryptionProtectorModel))]
+    public class SetAzureRmSqlManagedInstanceTransparentDataEncryptionProtector : AzureSqlRmManagedInstanceTransparentDataEncryptionProtectorBase
+    {
+        /// <summary>
+        /// Gets or sets the name of the Azure Sql Database Transparent Data Encryption protector type
+        /// </summary>
+        [Parameter(Mandatory = true,
+            ParameterSetName = InputObjectParameterSet,
+            Position = 1,
+            HelpMessage = "The Azure Sql Database Transparent Data Encryption Protector type.")]
+        [Parameter(Mandatory = true,
+            ParameterSetName = ResourceIdParameterSet,
+            Position = 1,
+            HelpMessage = "The Azure Sql Database Transparent Data Encryption Protector type.")]
+        [Parameter(Mandatory = true,
+            Position = 2,
+            ParameterSetName = DefaultParameterSet,
+            HelpMessage = "The Azure Sql Database Transparent Data Encryption Protector type.")]
+        [ValidateNotNullOrEmpty]
+        public EncryptionProtectorType Type { get; set; }
+
+        /// <summary>
+        /// Gets or sets the name of the Azure Sql Database server key vault KeyId
+        /// </summary>
+        [Parameter(Mandatory = false,
+            ParameterSetName = InputObjectParameterSet,
+            Position = 2,
+            HelpMessage = "The Azure Key Vault KeyId.")]
+        [Parameter(Mandatory = false,
+            ParameterSetName = ResourceIdParameterSet,
+            Position = 2,
+            HelpMessage = "The Azure Key Vault KeyId.")]
+        [Parameter(Mandatory = false,
+            ParameterSetName = DefaultParameterSet,
+            Position = 3,
+            HelpMessage = "The Azure Key Vault KeyId.")]
+        [ValidateNotNullOrEmpty]
+        public string KeyId { get; set; }
+        
+        /// <summary>
+        /// Defines whether it is ok to skip the requesting of setting Transparent Data Encryption protector confirmation
+        /// </summary>
+        [Parameter(HelpMessage = "Skip confirmation message for performing the action")]
+        public SwitchParameter Force { get; set; }
```

Comment:
I don't think you need a force here - force is reserved for cmdlets that make particularly destructive changes or have unexpected consequences (like deleting all the files when a folder is deleted)

Reply:
We require force here for Azure key vault protector because customer is adding azure key vault key as encryption protector. Thus the customer needs to understand that deleting this key will result in database being dropped. SQL Server cmdlets follow this pattern as well.